tool/tsh/common/access_request.go

```diff
@@ -406,6 +409,73 @@ func showRequestTable(cf *CLIConf, reqs []types.AccessRequest) error {
 }
 
 func onRequestSearch(cf *CLIConf) error {
+	if cf.RequestableRoles && cf.ResourceKind != "" {
+		return trace.BadParameter("only one of --kind and --roles may be specified")
+	}
+	if !cf.RequestableRoles && cf.ResourceKind == "" {
+		return trace.BadParameter("one of --kind and --roles is required")
+	}
+
+	if cf.RequestableRoles {
+		return searchRequestableRoles(cf)
+	} else {
+		return searchRequestableResources(cf)
+	}
+}
+
+func searchRequestableRoles(cf *CLIConf) error {
+	tc, err := makeClient(cf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var allRoles []*proto.ListRequestableRolesResponse_RequestableRole
+	err = tc.WithRootClusterClient(cf.Context, func(clt authclient.ClientI) error {
+		pageFunc := func(ctx context.Context, pageSize int, pageToken string) ([]*proto.ListRequestableRolesResponse_RequestableRole, string, error) {
+			req := &proto.ListRequestableRolesRequest{
+				PageSize:  int32(pageSize),
+				PageToken: pageToken,
+			}
+
+			resp, err := clt.ListRequestableRoles(ctx, req)
+			if err != nil {
+				return nil, "", trace.Wrap(err)
+			}
+
+			return resp.Roles, resp.NextPageToken, nil
```

Comment:
If we used the generated Getters which are nil safe we could eliminate a few lines here.
```suggestion
			resp, err := clt.ListRequestableRoles(ctx, req)
			return resp.GetRoles(), resp.GetNextPageToken(), trace.Wrap(err)
```

Reply:
I really like that, I’ll update the PR